### PR TITLE
Rahul/ifl 2101 proofgenerationkey serialization deserialization

### PR DESCRIPTION
**Pull request by patnir**
## Summary

Serialize + Deserialize Proof Generation Key.

## Testing Plan

Unit tests

## Documentation

Does this change require any updates to the Iron Fish Docs (ex. [the RPC API
Reference](https://ironfish.network/docs/onboarding/rpc/chain))? If yes, link a
related documentation pull request for the website.

```
[ ] Yes
```

## Breaking Change

Is this a breaking change? If yes, add notes below on why this is breaking and
what additional work is required, if any.

```
[ ] Yes
```

### CODE REVIEW

File: ironfish-zkp/src/primitives/proof_generation_key.rs [hunk 1]
```diff
@@ -0,0 +1,153 @@
...
+            deserialized_proof_generation_key.ak
+        );
+        assert_eq!(
+            proof_generation_key.nsk,
+            deserialized_proof_generation_key.nsk
+        );
+    }
+
+    #[test]
+    fn test_proof_generation_key_serializable_deserializable() {
```
Comment (hughy) [inline ironfish-zkp/src/primitives/proof_generation_key.rs:132]: does this test cover anything different than the one above it?

**Review by hughy — COMMENTED** [review]
small comment on possibly redundant test, but looks good otherwise!

File: ironfish-zkp/src/primitives/proof_generation_key.rs (hunk 1 above)
Comment (jowparks) [reply ironfish-zkp/src/primitives/proof_generation_key.rs:132]: Looks like same test, you can remove one. Otherwise looks good.

**Review by jowparks — APPROVED** [review]
Note that we checked the underlying library, there is no serde for this type. Also, we decided to put it in the `ironfish-zkp` because both zkp/rust sub projects rely on this. Since we don't have a lib, and `ironfish-zkp` is imported into `ironfish-rust` seemed like the only option.

File: ironfish-zkp/src/primitives/proof_generation_key.rs (hunk 1 above)
Comment (patnir) [reply ironfish-zkp/src/primitives/proof_generation_key.rs]: Removed!

**Pull request merged**